libs/game/physics.ts

```diff
@@ -352,9 +351,7 @@ class ArcadePhysicsEngine extends PhysicsEngine {
         // if the sprite is already clipping into a wall,
         // allow free movement rather than randomly 'fixing' it
         if (s.flags & sprites.Flag.IsClipping) {
-            if (tm.isOnWall(s)) {
```

Comment:
Technically this is a behavior change, but removing this portion makes it so we can still trigger overlaps with tiles if the sprite is clipping; I think this is the right behavior, no reason that a sprite being stuck in a wall should mean it doesn't take lava damage / etc